design/Extend-VolumePolicies-to-support-more-actions.md

```diff
@@ -0,0 +1,161 @@
+# Extend VolumePolicies to support more actions
+
+## Abstract
+
+Currently, the [VolumePolicies feature](https://github.com/vmware-tanzu/velero/blob/main/design/Implemented/handle-backup-of-volumes-by-resources-filters.md) which can be used to filter/handle volumes during backup only supports the skip action on matching conditions. Users need more actions to be supported.
+
+## Background
+
+The `VolumePolicies` feature was introduced in Velero 1.11 as a flexible way to handle volumes. The main agenda of introducing the VolumePolicies feature was to improve the overall user experience when performing backup operations for volume resources, the feature enables users to group volumes according the `conditions` (criteria) specified and also lets you specify the `action` that velero needs to take for these grouped volumes during the backup operation. The limitation being that currently `VolumePolicies` only supports `skip` as an action, We want to extend the `action` functionality to support more usable options like `fs-backup-volumes` (File system backup) and `fs-backup-volumes-excludes` (VolumeSnapshots).
+
+## Goals
+- Extending the VolumePolicies to support more actions like file system backup(`fs-backup-volumes`) and snapshots(`fs-backup-volumes-excludes`).
+- Improve user experience when backing up Volumes via Velero
+
+## Non-Goals
+- No changes to existing approaches to opt-in/opt-out annotations for volumes
+- No changes to existing `VolumePolicies` functionalities
+
+
+## Use-cases/Scenarios
+
+**Use-case 1:**
```

Comment:
@reasonerjt I"m not sure what you mean. If the user sets default-volumes-to-fs-backup, then that has higher priority, otherwise snapshot/CSI has higher priority.

I think replicating opt-in/opt-out is the only way to do this that makes sense unless we want to deprecate default-volumes-to-fs-backup and the whole opt-in/opt-out approach. Currently, that's the only supported way to decide between backup methods, so the most obvious way to go forward with this is to stick with that same model. The other approach would be to define a completely different workflow here and error out if user sets default-to-fs-backup *and* volume backup type policies -- but if we do that, then we basically have to start over with this design. @shubham-pampattiwar any thoughts?

Reply:
Resolving this comment as we agreed to use a more straight forward way to define the actions to the volumes.